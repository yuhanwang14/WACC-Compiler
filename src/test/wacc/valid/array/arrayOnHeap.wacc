# ensures that arrays are heap allocated

# Output:
# false
# 0
# 1
#

# Program:

begin
  int[] dummy_arr = [] ;
  int[][] arrs = [dummy_arr, dummy_arr] ;
  int i = 0;
  while i < 2 do
    int[] arr = [i] ;
    arrs[i] = arr ;
    i = i + 1
  done;

  # should have different addresses
  println arrs[0] == arrs[1];
  # should have different values
  println arrs[0][0];
  println arrs[1][0]
end
