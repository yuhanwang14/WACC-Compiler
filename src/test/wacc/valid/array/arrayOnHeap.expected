false
0
1

